frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts

```diff
@@ -501,72 +507,89 @@ export class MonitoredMetricsComponent implements OnInit, OnChanges, OnDestroy {
+    monitoredMetricsFormData.queries = monitoredMetricsFormData.queries.map(
+      (query, index) => {
+        let { keys, operationType, queryName, compareFn, compareValue, repeatedMeasure } = query;
+        if (keys) {
+          // check for metric key required except default row:
+          if ( keys[0].metricKey || operationType || queryName || compareFn || compareValue ) {
+            this.checkMetricKeyRequiredError(keys);
+          }
+          keys = keys.filter((key) => key.metricKey !== null).map(key => key.metricKey.key ? key.metricKey.key : key.metricKey);
+          if (keys.length) {
+            this.checkQueryNameRequiredError(queryName);
+            this.checkStatisticRequiredError(operationType);
+            let metric = this.allMetrics.find(metric => metric.key === keys[0]);
+            const { metadata: { type } } = metric;
+            if (type === IMetricMetaData.CATEGORICAL) {
+              this.checkComparisonStatisticRequiredError(compareFn, compareValue);
+            }
+            let queryObj: Query = {
+              name: queryName,
+              query: {
+                operationType
+              },
+              metric: {
+                key: keys.join(METRICS_JOIN_TEXT)
+              },
+              repeatedMeasure
+            };
+            if (compareFn && !!compareValue) {
+              queryObj = {
+                ...queryObj,
+                query: {
+                  ...queryObj.query,
+                  compareFn,
+                  compareValue
                 }
-                return this.experimentInfo
-                  ? ({ ...this.experimentInfo.queries[index], ...queryObj })
-                  : (queryObj.metric.key 
-                    ? ({...queryObj}) 
-                    : ({...this.removeMetricName(queryObj)})
-                );
               }
             }
+            return this.experimentInfo
+              ? ({ ...this.experimentInfo.queries[index], ...queryObj })
+              : (queryObj.metric.key 
+                ? ({...queryObj}) 
+                : ({...this.removeMetricName(queryObj)})
+            );
           }
-        );
-
-        if (this.queryMetricKeyError.length === 0 && this.queryStatisticError.length === 0 && this.queryComparisonStatisticError.length === 0 && this.queryNameError.length === 0) {
-          this.emitExperimentDialogEvent.emit({
-            type: eventType,
-            formData: monitoredMetricsFormData,
-            path: NewExperimentPaths.MONITORED_METRIC
-          });
-          break;
         }
+      }
+    );
+
+    if (this.queryMetricKeyError.length === 0 && this.queryStatisticError.length === 0 && this.queryComparisonStatisticError.length === 0 && this.queryNameError.length === 0) {
+      this.emitExperimentDialogEvent.emit({
+        type: eventType,
+        formData: monitoredMetricsFormData,
+        path: NewExperimentPaths.MONITORED_METRIC
+      });
     }
```

Comment:
looks like a `break` statement is missing

Reply:
saveData() function is called in switch case and there are break statement after function called there after flag reset.